# Movie Oscar win Prediction

![](oscar1.jpg)

In [1]:
# Importing Libraries
import numpy as np
import pandas as pd

import warnings
warnings.filterwarnings('ignore')


from sklearn.ensemble import RandomForestClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.naive_bayes import GaussianNB
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder
from sklearn.metrics import f1_score


# from time import time
# from os import path, makedirs, walk
# from joblib import dump, load
# import json

In [2]:
data = pd.read_csv('../DataSets/Movie_classification.csv')
data.head()

Marketing expense  Production expense  Multiplex coverage     Budget  \
0            20.1264               59.62               0.462  36524.125   
1            20.5462               69.14               0.531  35668.655   
2            20.5458               69.14               0.531  39912.675   
3            20.6474               59.36               0.542  38873.890   
4            21.3810               59.36               0.542  39701.585   

   Movie_length  Lead_ Actor_Rating  Lead_Actress_rating  Director_rating  \
0         138.7               7.825                8.095            7.910   
1         152.4               7.505                7.650            7.440   
2         134.6               7.485                7.570            7.495   
3         119.3               6.895                7.035            6.920   
4         127.7               6.920                7.070            6.815   

   Producer_rating  Critic_rating  Trailer_views 3D_available  Time_taken  \
0            7.995           7.94         527367          YES      109.60   
1            7.470           7.44         494055           NO      146.64   
2            7.515           7.44         547051           NO      147.88   
3            7.020           8.26         516279          YES      185.36   
4            7.070           8.26         531448           NO      176.48   

   Twitter_hastags     Genre  Avg_age_actors  Num_multiplex  Collection  \
0          223.840  Thriller              23            494       48000   
1          243.456     Drama              42            462       43200   
2         2022.400    Comedy              38            458       69400   
3          225.344     Drama              45            472       66800   
4          225.792     Drama              55            395       72400   

   Start_Tech_Oscar  
0                 1  
1                 0  
2                 1  
3                 1  
4                 1

**Finding out the correlation between the attributes of the dataset**

In [3]:
data.corr()

Marketing expense  Production expense  \
Marketing expense             1.000000            0.406583   
Production expense            0.406583            1.000000   
Multiplex coverage           -0.420972           -0.763651   
Budget                       -0.219247           -0.391676   
Movie_length                  0.352734            0.644779   
Lead_ Actor_Rating            0.380050            0.706481   
Lead_Actress_rating           0.379813            0.707956   
Director_rating               0.380069            0.707566   
Producer_rating               0.376462            0.705819   
Critic_rating                -0.184985           -0.251565   
Trailer_views                -0.443457           -0.591657   
Time_taken                    0.026019            0.015888   
Twitter_hastags               0.013518           -0.000839   
Avg_age_actors                0.059204            0.055810   
Num_multiplex                 0.383298            0.707559   
Collection                   -0.389582           -0.484754   
Start_Tech_Oscar             -0.013417           -0.024404   

                     Multiplex coverage    Budget  Movie_length  \
Marketing expense             -0.420972 -0.219247      0.352734   
Production expense            -0.763651 -0.391676      0.644779   
Multiplex coverage             1.000000  0.302188     -0.731470   
Budget                         0.302188  1.000000     -0.240265   
Movie_length                  -0.731470 -0.240265      1.000000   
Lead_ Actor_Rating            -0.768589 -0.208464      0.746904   
Lead_Actress_rating           -0.769724 -0.203981      0.746493   
Director_rating               -0.769157 -0.201907      0.747021   
Producer_rating               -0.764873 -0.205397      0.746707   
Critic_rating                  0.145555  0.232361     -0.217830   
Trailer_views                  0.581386  0.602536     -0.589318   
Time_taken                     0.035922  0.040773     -0.019984   
Twitter_hastags                0.004882  0.030674      0.009380   
Avg_age_actors                -0.092104 -0.064694      0.075198   
Num_multiplex                 -0.915495 -0.282796      0.673896   
Collection                     0.429300  0.696304     -0.377999   
Start_Tech_Oscar              -0.004017 -0.027148      0.016291   

                     Lead_ Actor_Rating  Lead_Actress_rating  Director_rating  \
Marketing expense              0.380050             0.379813         0.380069   
Production expense             0.706481             0.707956         0.707566   
Multiplex coverage            -0.768589            -0.769724        -0.769157   
Budget                        -0.208464            -0.203981        -0.201907   
Movie_length                   0.746904             0.746493         0.747021   
Lead_ Actor_Rating             1.000000             0.997905         0.997735   
Lead_Actress_rating            0.997905             1.000000         0.998097   
Director_rating                0.997735             0.998097         1.000000   
Producer_rating                0.994073             0.994003         0.994126   
Critic_rating                 -0.169978            -0.165992        -0.166638   
Trailer_views                 -0.490267            -0.487536        -0.486452   
Time_taken                     0.038494             0.038432         0.036300   
Twitter_hastags                0.014463             0.010239         0.010077   
Avg_age_actors                 0.036794             0.038005         0.041470   
Num_multiplex                  0.706331             0.708257         0.709364   
Collection                    -0.251355            -0.249459        -0.246650   
Start_Tech_Oscar              -0.035309            -0.040356        -0.035768   

                     Producer_rating  Critic_rating  Trailer_views  \
Marketing expense           0.376462      -0.184985      -0.443457   
Production expense          0.705819      -0.251565      -0.591657   
Multiplex coverage         -

In [4]:
data.head()

Marketing expense  Production expense  Multiplex coverage     Budget  \
0            20.1264               59.62               0.462  36524.125   
1            20.5462               69.14               0.531  35668.655   
2            20.5458               69.14               0.531  39912.675   
3            20.6474               59.36               0.542  38873.890   
4            21.3810               59.36               0.542  39701.585   

   Movie_length  Lead_ Actor_Rating  Lead_Actress_rating  Director_rating  \
0         138.7               7.825                8.095            7.910   
1         152.4               7.505                7.650            7.440   
2         134.6               7.485                7.570            7.495   
3         119.3               6.895                7.035            6.920   
4         127.7               6.920                7.070            6.815   

   Producer_rating  Critic_rating  Trailer_views 3D_available  Time_taken  \
0            7.995           7.94         527367          YES      109.60   
1            7.470           7.44         494055           NO      146.64   
2            7.515           7.44         547051           NO      147.88   
3            7.020           8.26         516279          YES      185.36   
4            7.070           8.26         531448           NO      176.48   

   Twitter_hastags     Genre  Avg_age_actors  Num_multiplex  Collection  \
0          223.840  Thriller              23            494       48000   
1          243.456     Drama              42            462       43200   
2         2022.400    Comedy              38            458       69400   
3          225.344     Drama              45            472       66800   
4          225.792     Drama              55            395       72400   

   Start_Tech_Oscar  
0                 1  
1                 0  
2                 1  
3                 1  
4                 1

**Information about the dataset**

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 506 entries, 0 to 505
Data columns (total 19 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Marketing expense    506 non-null    float64
 1   Production expense   506 non-null    float64
 2   Multiplex coverage   506 non-null    float64
 3   Budget               506 non-null    float64
 4   Movie_length         506 non-null    float64
 5   Lead_ Actor_Rating   506 non-null    float64
 6   Lead_Actress_rating  506 non-null    float64
 7   Director_rating      506 non-null    float64
 8   Producer_rating      506 non-null    float64
 9   Critic_rating        506 non-null    float64
 10  Trailer_views        506 non-null    int64  
 11  3D_available         506 non-null    object 
 12  Time_taken           494 non-null    float64
 13  Twitter_hastags      506 non-null    float64
 14  Genre                506 non-null    object 
 15  Avg_age_actors       506 non-null    int

In [6]:
data.columns

Index(['Marketing expense', 'Production expense', 'Multiplex coverage',
       'Budget', 'Movie_length', 'Lead_ Actor_Rating', 'Lead_Actress_rating',
       'Director_rating', 'Producer_rating', 'Critic_rating', 'Trailer_views',
       '3D_available', 'Time_taken', 'Twitter_hastags', 'Genre',
       'Avg_age_actors', 'Num_multiplex', 'Collection', 'Start_Tech_Oscar'],
      dtype='object')

**Creating the dataset for the training and testing of the model**

In [7]:
x = data[['Collection','Twitter_hastags']]
y = data['Start_Tech_Oscar']

x_train, x_test, y_train, y_test = train_test_split(x,y, test_size = 0.2)

## Prediction Model Creation


### Logistic Regression 

Logistic regression is a statistical model that in its basic form uses a logistic function to model a binary dependent variable, although many more complex extensions exist. In regression analysis, logistic regression (or logit regression) is estimating the parameters of a logistic model (a form of binary regression).

In [8]:
logReg = LogisticRegression(max_iter = 1000)
logReg.fit(x_train, y_train)

LogisticRegression(max_iter=1000)

In [9]:
logReg.score(x_test, y_test)

0.5098039215686274

### Decision Tree Classifier Algorithm

In [10]:
from sklearn.tree import DecisionTreeClassifier
dtc = DecisionTreeClassifier()
dtc.fit(x_train, y_train)
dtc.score(x_test, y_test)

0.5196078431372549

### Random Forest Classifier Algorithm


In [11]:
rfc = RandomForestClassifier()
rfc.fit(x_train, y_train)
rfc.score(x_test, y_test)

0.5098039215686274

### K-Nearest Neighbours Algorithm


In [12]:
from sklearn.neighbors import KNeighborsClassifier  
classifier= KNeighborsClassifier(n_neighbors=5, metric='minkowski', p=2 ) 

In [13]:
classifier.fit(x_train, y_train)  

KNeighborsClassifier()

In [14]:
classifier.score(x_test, y_test)

0.47058823529411764

### Gausian NB Algorithm


In [15]:
clf = GaussianNB()
clf.fit(x_train, y_train)

GaussianNB()

In [16]:
clf.score(x_test, y_test)

0.5294117647058824

### Support Vector Machine Algorithm

In [17]:
svm = SVC()
svm.fit(x_train, y_train)

SVC()

In [18]:
svm.score(x_test, y_test)

0.5098039215686274

### Gradient Boosting

In [19]:
from sklearn.ensemble import GradientBoostingClassifier
clf = GradientBoostingClassifier(n_estimators=150, learning_rate=3.0,max_depth=0.5, random_state=0)

In [20]:
clf.fit(x_train, y_train)

GradientBoostingClassifier(learning_rate=3.0, max_depth=0.5, n_estimators=150,
                           random_state=0)

In [21]:
clf.score(x_test, y_test)

0.5098039215686274

### MLP Classifier

In [22]:
from sklearn.neural_network import MLPClassifier
clf = MLPClassifier(random_state=2, max_iter=300)

In [23]:
clf.fit(x_train, y_train)

MLPClassifier(max_iter=300, random_state=2)

In [24]:
clf.score(x_test, y_test)

0.5098039215686274

### Stochastic Gradient Descent (SGD)

In [25]:
from sklearn.linear_model import SGDClassifier
sdg = SGDClassifier()


In [26]:
sdg.fit(x_train, y_train)

SGDClassifier()

In [27]:
sdg.score(x_test, y_test)

0.49019607843137253

### Linear Discriminant Analysis (LDA).

In [28]:
from sklearn.discriminant_analysis import LinearDiscriminantAnalysis
lda = LinearDiscriminantAnalysis()

In [29]:
lda.fit(x_train, y_train)

LinearDiscriminantAnalysis()

In [30]:
lda.score(x_test, y_test)

0.5

**********

### Model  Scores
|Name of the Model|Accuracy Score|
|:---:|:---:|
|Logistic Regression|0.60|
|Decision Tree Classifier|0.47|
|Random Forest Classifier|0.51|
|Gausian NB Algorithm|0.50|
|KNN Algorithm|0.61|
|Support Vector Machine Algorithm|0.50|
|Gradient Boosting|0.64|
|MLP Classifier|0.64|
|Stochastic Gradient Descent|0.64|
|Linear Discriminant Analysis|0.51|


